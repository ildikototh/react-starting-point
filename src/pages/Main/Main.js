import React from 'react';
import Header from "../../layouts/Header";
import Footer from "../../layouts/Footer";

const Main = () => (
  <div className="App">
    <Header />
    <div className="content">React Starting point boilerplate</div>
    <Footer />
  </div>
);

export default Main;
