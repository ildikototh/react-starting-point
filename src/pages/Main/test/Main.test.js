import React from 'react';
import { shallow } from 'enzyme';
import Main from '../Main';

describe('<Main />', () => {
  describe('structure', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Main />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render  <Header />', () => {
      expect(wrapper).toContainMatchingElement('Header');
    });

    it('should render  <Footer />', () => {
      expect(wrapper).toContainMatchingElement('Footer');
    });

    it('should contain div box with \'React Starting point boilerplate\' text', () => {
      expect(wrapper.find('.content').text()).toEqual('React Starting point boilerplate');
    });
  });
});
