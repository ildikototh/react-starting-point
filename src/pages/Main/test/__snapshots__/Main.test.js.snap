// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> structure should match the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <div
    className="content"
  >
    React Starting point boilerplate
  </div>
  <Footer />
</div>
`;
